sycl/include/CL/sycl/detail/pi.hpp

```diff
@@ -120,45 +120,68 @@ void printArgs(Arg0 arg0, Args... args) {
 // Utility function to check return from pi calls.
 // Throws if pi_result is not a PI_SUCCESS.
 template <typename Exception = cl::sycl::runtime_error>
-inline void piCheckResult(PiResult pi_result) {
+inline void checkPiResult(PiResult pi_result) {
   CHECK_OCL_CODE_THROW(pi_result, Exception);
 }
 
-#define PI_TRACE_ENABLED (std::getenv("SYCL_PI_TRACE") != nullptr)
-
-template <typename FnType, size_t FnOffset> class Trace {
+// Class to call PI API, trace and get the result.
+// To Trace : Set SYCL_PI_TRACE environment variable.
+// Template Arguments:
+//    FnType  - Type of Function pointer to the PI API.
+//    FnOffset- Offset to the Function Pointer in the piPlugin::FunctionPointers
+//    structure.Used to differentiate between APIs with same pointer type, Eg:
+//    piDeviceRelease and piDeviceRetain. Differentiation needed to avoid
+//    duplicate instantiation of class in pi.cpp.
```

Comment:
please clarify that this is really a stability issue, i.e. you cannot do different template specializations for different PI interfaces (having same type) without this